browser/js/dashboard/gridbox/widgetSettingsCtrl.js

```diff
@@ -16,39 +16,20 @@
         type: widget.type,
         dataSource: widget.dataSource,
         yparam: widget.yparam,
-        xparam: widget.xparam
+        xparam: widget.xparam,
+        refreshInterval : widget.refreshInterval || 3000,
       };
+      //3000 is a placeholder default value for testing, should switch to higher value when deploying. Also we need to put refreshInterval in the form logic
 
 
     let dataInNVD3Format;
      $scope.setKeys = function(){
-      DashboardFactory.getDataSource($scope.form.dataSource)
-      .then(function(data){
-        let realData = findDataToGraph(data);
-        $scope.dataKeys = Object.keys(realData[0]);
-        dataInNVD3Format = [{
-          values:realData,
-          key: "this works",
-          color: '#ff7f0e'
-        }];
-        widget.chart.data = dataInNVD3Format;
-      })
-
-     }
-
-  function findDataToGraph(obj){
-    if(Array.isArray(obj)){
-      return obj;
-    }else if(typeof(obj) === 'object'){
-      var x;
-      for(var key in obj){
-       x = findDataToGraph(obj[key]);
-       if(x){
-        return x;
-       }
+        return WidgetSettingsFactory.newSetKeys($scope.form.dataSource)
```

Comment:
Watch out for returning things in your `$scope` methods. These are used in the template, and so if you return something here it may end up being passed to the view somehow—in this case the view will probably not be directly concerned with the promise this comes back with.